rules-reviewed/azure/eap/eap-to-azure-appservice-environment-variables.windup.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0"?>
+<ruleset id="eap-to-azure-appservice-environment-variables"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Validate that all usages of System.getenv and System.getProperty are captured
+            as App Settings when migrating to Azure App Service.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[7,8)" />
+        <targetTechnology id="azure-appservice"/>
+        <tag>JBoss EAP</tag>
+        <tag>Azure</tag>
+    </metadata>
+    <rules>
+        <rule id="eap-to-azure-appservice-environment-variables-001">
+            <when>
+                <or>
+                    <!-- as compiled code -->
+                    <javaclass references="java.lang.System.getenv({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <javaclass references="java.lang.System.getProperty({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <!-- as source code -->
+                    <filecontent filename="{*}.java" pattern="System.getenv"/>
+                    <filecontent filename="{*}.java" pattern="System.getProperty"/>

```

Comment:
```suggestion
```
These lines can be removed because the above `javaclass` conditions work in both compiled and source mode